8344337: SecurityManager cleanup in java.prefs module

bchristi-git opened the pull request:
Remove usages of SecurityManager, doPrivildged, and AccessController from the java.prefs module.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344337](https://bugs.openjdk.org/browse/JDK-8344337): SecurityManager cleanup in java.prefs module (**Bug** - P4)


### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22252/head:pull/22252` \
`$ git checkout pull/22252`

Update a local copy of the PR: \
`$ git checkout pull/22252` \
`$ git pull https://git.openjdk.org/jdk.git pull/22252/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22252`

View PR using the GUI difftool: \
`$ git pr show -t 22252`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22252.diff">https://git.openjdk.org/jdk/pull/22252.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22252#issuecomment-2486507273)
</details>

LanceAndersen commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -27,9 +27,9 @@
+//import java.security.AccessController;
+//import java.security.PrivilegedAction;
+//import java.security.PrivilegedExceptionAction;
 import java.security.PrivilegedActionException;
```
I think you can remove these vs leaving them commented out

LanceAndersen commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -53,22 +53,21 @@ class FileSystemPreferences extends AbstractPreferences {
+//            return null;
+//        };
+//        AccessController.doPrivileged(load);
     }
```
Looks like there is some clean up here to do of the commented out code...

bchristi-git replied:
Whoops!

LanceAndersen reviewed (changes requested):
Hi Brent,

There looks to be a lot of commented out code that should be removed as part of your PR?

seanjmullan commented:
A few classes need updated copyrights.

RogerRiggs commented on src/java.prefs/share/classes/java/util/prefs/Preferences.java:
```diff
@@ -267,11 +245,7 @@ public String run() {
-            new PrivilegedAction<PreferencesFactory>() {
-                public PreferencesFactory run() {
-                    return factory1();}});
+        return factory1();
```
No need to make factory1 a separate method, merge into factory.

RogerRiggs commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -765,56 +735,35 @@ public synchronized void sync() throws BackingStoreException {
-        } catch (PrivilegedActionException e) {
-            throw (BackingStoreException) e.getException();
-        }
+        syncSpiPrivileged();
```
Inline syncSpiPrivileged(), it doesn't need to be separate and the name "privileged" is misleading

bchristi-git commented on src/java.prefs/share/classes/java/util/prefs/Preferences.java:
```diff
@@ -30,20 +30,10 @@
 import java.util.ServiceLoader;
 import java.util.ServiceConfigurationError;
 
-// These imports needed only as a workaround for a JavaDoc bug
```
This comment was added quite a while ago. But FWIW, a specdiff comparison shows no JavaDoc differences.

bchristi-git commented:
> I think the try-block at line 137 in FileSystemPreferences would be more readable if it were indented. I know it was not indented in the original, either.

Oh, yeah, I don't like that either. :(
Fixed; also added {}s.
Thanks.

bplb commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -117,52 +109,47 @@ static Preferences getUserRoot() {
+                getLogger().info("Created user preferences directory.");
             }
-        });
+            else
```
It seems odd that `else` is not on line 125 after the bracket. If putting it on the same line makes it  less readable, then maybe add a pair of brackets for the `else`.

bplb reviewed (approved):
Looks all right to me now, but you might want also to get a second opinion.

turbanoff commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -117,52 +109,47 @@ static Preferences getUserRoot() {
+        }
+        isUserRootWritable = userRootDir.canWrite();
+        String USER_NAME = System.getProperty("user.name");
+        userLockFile = new File (userRootDir,".user.lock." + USER_NAME);
```
```suggestion
        userLockFile = new File(userRootDir, ".user.lock." + USER_NAME);
```

turbanoff commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -185,58 +172,53 @@ static Preferences getSystemRoot() {
-                                + "preferences are unusable.");
-                        }
+        String systemPrefsDirName =
+          System.getProperty("java.util.prefs.systemRoot","/etc/.java");
```
```suggestion
          System.getProperty("java.util.prefs.systemRoot", "/etc/.java");
```

turbanoff commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -185,58 +172,53 @@ static Preferences getSystemRoot() {
+        isSystemRootWritable = systemRootDir.canWrite();
+        systemLockFile = new File(systemRootDir, ".system.lock");
+        systemRootModFile =
+                       new File (systemRootDir,".systemRootModFile");
```
```suggestion
                       new File(systemRootDir, ".systemRootModFile");
```

turbanoff commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -596,43 +566,32 @@ private void initCacheIfNecessary() {
+            try (FileInputStream fis = new FileInputStream(prefsFile)) {
+                XmlSupport.importMap(fis, m);
+            }
+        } catch(Exception e) {
```
```suggestion
        } catch (Exception e) {
```

bchristi-git replied:
Most catch statements in this file do not have a space.

turbanoff commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -185,58 +172,53 @@ static Preferences getSystemRoot() {
+        systemLockFile = new File(systemRootDir, ".system.lock");
+        systemRootModFile =
+                       new File (systemRootDir,".systemRootModFile");
+        if (!systemRootModFile.exists() && isSystemRootWritable)
```
let's add braces here too

RogerRiggs commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -117,52 +109,47 @@ static Preferences getUserRoot() {
-                if (!userRootModFile.exists())
+        userRootDir =
+              new File(System.getProperty("java.util.prefs.userRoot",
+              System.getProperty("user.home")), ".java/.userPrefs");
```
This expression would be more readable if the 2nd line was indented.

RogerRiggs commented on src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:
```diff
@@ -466,16 +447,11 @@ public void run() {
-                });
-                return null;
+        Runtime.getRuntime().addShutdownHook(
+            new Thread(null, null, "Sync Timer Thread", 0, false) {
```
Future work, use a Virtual thread for this.

bchristi-git replied:
https://bugs.openjdk.org/browse/JDK-8344793

RogerRiggs reviewed (approved):
Looks good

LanceAndersen reviewed (approved):
Latest updates look good

bplb reviewed (approved):
+1

irisclark commented on src/java.prefs/macosx/classes/java/util/prefs/MacOSXPreferencesFile.java:
```diff
@@ -82,15 +82,9 @@ class MacOSXPreferencesFile {
-                    return null;
-                }
-            });
+        System.loadLibrary("prefs");
```
What am I missing?  Why keep the private static method if it only contains a single method invocation and is only used in the static initializer?

bchristi-git replied:
We want to `@SuppressWarnings("restricted")` the `loadLibrary()` call. IIUC, I can't `@SuppressWarnings` the static block, nor the call to `loadLibrary()`, so AFAICT that call needs to be in a separate, `@SuppressWarnings`-ed method.

seanjmullan replied:
Right. The other option is to put the annotation above the class definition which would cover the entire class, but I think this is better as it makes the annotation specific to that method only.

bchristi-git commented:
/integrate
